mon-pix/app/controllers/assessments/checkpoint.js

```diff
@@ -5,15 +5,32 @@ export default Controller.extend({
 
   queryParams: ['finalCheckpoint'],
   finalCheckpoint: false,
+  isShowingModal: false,
+  answer: null,
+  correction: null,
+  challenge: null,
 
   nextPageButtonText: computed('finalCheckpoint', function() {
     return this.get('finalCheckpoint') ? 'Voir mes résultats' : 'Continuer mon parcours';
   }),
 
   actions: {
-    openComparison(assessment_id, answer_id, index) {
-      return this.transitionToRoute('assessments.comparison', assessment_id, answer_id, index);
-    }
+    async openComparisonWindow(answerId) {
+      const store = this.get('store');
+
+      const answer = await store.findRecord('answer', answerId);
+      this.set('answer', answer);
+      const challenge = await answer.get('challenge');
+      this.set('challenge', challenge);
+      const correction = await store.query('correction', { answerId }).then((corrections) => corrections.get('firstObject'));
```

Comment:
Assez bizarre toute cette mécanique. Il faudrait seulement récupérer l'answer ici avec le challenge et la correction soit dans un include soit dans des links.
Qui plus est j'ai l'impression qu'on a déjà l'answer normalement, pourquoi la récupérer à nouveau ?

Reply:
Je viens de tester en ne forçant pas le refresh de l'answer, et en utilisant directement celle récupérée précédemment : ça plante (les infos relatives au résultat, dans le header du détail, ne s'affichent pas) car actuellement, quand on récupère une réponse - `GET /api/answers/:answer_id` - elle ne contient pas les infos de résultat. 

Je continue de checker si ont peut améliorer la séquence d'appels à moindre coût (sachant que j'aime pas l'idée de toucher à l'API dans une tâche purement orientée front, qui plus est dont le but premier était de permettre la montée de version de l'appli).

Reply:
Il y a quand même un truc très très étrange : sur la liste on a le résultat pour chaque answers, mais pas dans le header de la comparison-window... 🤔

Reply:
J'ai compris, dans le controller de la route, on ne fait pas que récupérer l'answer, on l'asssign au contexte de la route, afin que cette propriété soit celle utilisée par la `{{pix-modal}}`. Facile :-)

Reply:
On n'est pas non plus obligé de récupérer le challenge indépendamment dans le controller :-)

Reply:
Moralité : 
- on ne récupère plus l'answer à l'ouverture de la comparison-window (récupérée dans assessments/results)
- on ne récupère plus non plus le challenge (récupérée dans assessments/results)
- on continue de demander la correction
- on continue de faire beaucoup trop d'appels (1 + 2*answers)